h2gis-functions/src/main/java/org/h2gis/functions/io/geojson/GeoJsonReaderDriver.java

```diff
@@ -1278,4 +1297,100 @@ private void setGeometryTypeConstraints() throws SQLException {
             connection.createStatement().execute(String.format("ALTER TABLE %s ALTER COLUMN the_geom SET DATA TYPE geometry(%s,%d)", tableLocation.toString(), finalGeometryType, parsedSRID));
         }
     }
+
+
+    /**
+     * Parses Json Array.
+     * Syntax:
+     * Json Array:
+     * {"member1": value1}, value2, value3, {"member4": value4}]
+     * @author Pham Hai Trung
+     * @param jp the json parser
+     * @return the array but written like a String
+     */
+    private void parseArrayMetadata(JsonParser jp) throws IOException {
+        JsonToken value = jp.nextToken();
+        while(value != JsonToken.END_ARRAY) {
+            if (value == JsonToken.START_OBJECT) {
+                parseObjectMetadata(jp);
+            } else if (value == JsonToken.START_ARRAY) {
+                parseArrayMetadata(jp);
+            }
+            value = jp.nextToken();
+        }
+    }
+
+    /**
+     * Parses Json Object.
+     * Syntax:
+     * Json Object:
+     * "member1": value1, "member2": value2}
+     * @author Pham Hai Trung
+     * @param jp the json parser
+     * @return the object but written like a String
+     */
+    private void parseObjectMetadata(JsonParser jp) throws IOException {
+        JsonToken value;
+        while (jp.nextToken() != JsonToken.END_OBJECT) {
+            value = jp.nextToken();
+            if (value == JsonToken.START_OBJECT) {
+                parseObjectMetadata(jp);
+            } else if (value == JsonToken.START_ARRAY) {
+                parseArrayMetadata(jp);
+            }
+        }
+    }
+
+    /**
+     * Parses Json Array and returns an ArrayList
+     * Syntax:
+     * Json Array:
+     * {"member1": value1}, value2, value3, {"member4": value4}]
+     * @param jp the json parser
+     * @return the array
+     */
+    private ArrayList<Object> parseArray(JsonParser jp) throws IOException {
+        JsonToken value = jp.nextToken();
+        ArrayList<Object> ret = new ArrayList<>();
+        while(value != JsonToken.END_ARRAY) {
+            if (value == JsonToken.START_OBJECT) {
+                Object object = parseObject(jp);
+                ret.add(object);
+            } else if (value == JsonToken.START_ARRAY) {
+                ArrayList<Object> array = parseArray(jp);
+                ret.add(array);
+            } else {
+                ret.add(jp.getValueAsString());
```

Comment:
What if the `jp` value is not a `String` bit a `int` for example? Maybe you should take the type into account.